maven_iuvs/echelle.py

```diff
@@ -1607,34 +1634,41 @@ def fit_flat_data(light_fits, spectrum, data_unc, bad_frames=None,
     fit_params_dicts = []
     fit_unc_dicts = []
 
-     # Loop over integrations to do the fits
+    # Initial guesses for every integration. Shape: [n_ints, num_params]
+    initial_guesses = line_fit_initial_guess(light_fits, wavelengths, spectrum)
+
+    # Loop over integrations to do the fits
     for i in range(0, ints_to_fit):
         if bad_frames:
             if i in bad_frames:
                 fit_params_dicts.append({"failed_fit": True, 
-                                         "area_H": 0, "area_D": 0, 
-                                         "area_IPH": 0, "lambdac_H": 121.567, 
-                                         "lambdac_IPH": 121.55, "width_IPH": 0, 
-                                         "M": 0, "B": 0})
-                fit_unc_dicts.append({"unc_H": 0, "unc_D": 0, 
-                                      "unc_IPH": 0, "unc_lambdac_H": 0, 
-                                      "unc_lambdac_IPH": 0, "unc_width_IPH": 0, 
-                                      "unc_M": 0, "unc_B": 0})
+                                         'total_brightness_H': 0, 'total_brightness_D': 0, 
+                                         'total_brightness_IPH': 0, 'central_wavelength_H': 121.567, 
+                                         'central_wavelength_IPH': 121.55, "width_IPH": 0, 
+                                         "background_m": 0, "background_b": 0})
+                fit_unc_dicts.append({'unc_total_brightness_H': 0, 'unc_total_brightness_D': 0, 
+                                      'unc_total_brightness_IPH': 0, 'unc_central_wavelength_H': 0, 
+                                      'unc_central_wavelength_IPH': 0, "unc_width_IPH": 0, 
+                                      'unc_background_m': 0, 'unc_background_b': 0})
                 continue # we already filled the arrays with zeros so just skip the other tasks
 
         # PERFORM FIT
         # ============================================================================================
         # Through experimentation, we found that the best solvers to use are in descending order: 
         # Powell, Nelder-Mead, and then TNC, CG, L-BFGS-B,and trust-constr are all kinda similar
-        initial_guess = line_fit_initial_guess(wavelengths, spectrum[i, :])
+        initial_guess = initial_guesses[i, :]
 
         if BU_bg is not None:
             BU_bg_i = BU_bg[i, :]
         else:
             BU_bg_i = None
 
-        result_vec = fit_H_and_D(initial_guess, wavelengths, spectrum[i, :], light_fits, theCLSF, unc=data_unc[i, :], BU_bg=BU_bg_i, **kwargs)
-            
+        # Determine whether line-of-sight minimum ray height is large enough to fit an IPH component
+        fit_IPH_component = check_whether_IPH_fittable(light_fits)
```

Comment:
technically this is duplicating work because the full FITS file is checked on every integration loop, but eh

Reply:
Fixed in 1dc86bfe73b86b1aad026fa19d45487656e827a4

Reply:
I think the new implementation is still redoing all of the work in every loop, but this is probably not an important optimization to make.

Reply:
Yes this seems to have been an oversight in trying to adapt to quicklooks. it has been fixed for real this time in 9756bee44f864647d52fa08bab306de82ab6bdc3